src/backend/vulkan/src/device.rs

```diff
@@ -1891,11 +1891,27 @@ impl d::Device<B> for super::Device {
     }
 
     fn start_capture(&self) {
-        //TODO: RenderDoc
+        #[cfg(feature = "renderdoc")]
+        unsafe {
+            ((*self.shared.instance.render_doc_entry.lock().0)
```

Comment:
do we actually need to lock it?. Is calling `StartFrameCapture` requiring `&mut self`?

Reply:
Nope, I don't think so. I had to unsafe impl Sync+Send on the `RenderDocEntry` though.

The [renderdoc.rs code](https://github.com/ebkalderon/renderdoc-rs/blob/560731b184b784725f6c00e32c13f46764a13ec7/src/version.rs#L62) mentions some thread safety issues when loading the entry point itself. I have scanned the renderdoc docs, but I haven't found anything obvious regarding thread safety issues.

Reply:
ok, in this case, let's not wrap it into a mutex, even. Just store it plan on the instance